src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```
@@ -1323,7 +1323,7 @@ because xbuild doesn't support framework reference assemblies.
 <Target Name="_ManagedUpdateAndroidResgen" Condition=" '$(ManagedDesignTimeBuild)' == 'True' "
 		Inputs="$(_ManagedUpdateAndroidResgenInputs);$(_AndroidResourcePathsCache);$(_AndroidLibraryProjectImportsCache);$(_AndroidLibraryImportsCache);"
 		Outputs="$(_AndroidManagedResourceDesignerFile)"
-		DependsOnTargets="_CreatePropertiesCache;_ExtractLibraryProjectImports;_CreateAdditionalResourceCache;_ValidateAndroidPackageProperties">
+		DependsOnTargets="_CreatePropertiesCache;_ExtractLibraryProjectImports;_AddAndroidDefines;_CreateAdditionalResourceCache;_ValidateAndroidPackageProperties">
```

This is a tad confusing, in that the stated purpose is to have `$(AndroidNdkDirectory)` set.  This should thus insert `$(_OnResolveMonoAndroidSdks)` instead of `_AddAndroidDefines`, because `_AddAndroidDefines` *implies* that that the reason it's in there is for "Android Defines" (e.g. the `__ANDROID__` define), when *that isn't what is intended at all*.  "Add Android Defines" is thus *misleading*, only being used because it will *implicitly* include `$(_OnResolveMonoAndroidSdks)`.

---

@adalon if you could rebase this on latest master so we have #3430, and change this to:
```suggestion
		DependsOnTargets="_CreatePropertiesCache;_ExtractLibraryProjectImports;_ResolveMonoAndroidSdks;_CreateAdditionalResourceCache;_ValidateAndroidPackageProperties">
```
We'll be all set.